services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +43,41 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
+  if (xQueueSend(thermalMgrQueueHandle, event, 1) != pdTRUE) return ERR_CODE_QUEUE_FULL; // timeout?
 
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t event;
+  event.type = THERMAL_MGR_EVENT_INTERRUPT;
+  if (thermalMgrSendEvent(&event) != ERR_CODE_SUCCESS) ; // What should be done if error?
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  lm75bd_config_t config = *(lm75bd_config_t *)pvParameters;
+  thermal_mgr_event_t event;
+  float temp;
+
   while (1) {
-    
+    if (xQueueReceive(thermalMgrQueueHandle, &event, 0) == pdTRUE) { // timeout?
+      switch (event.type) {
+        case THERMAL_MGR_EVENT_MEASURE_TEMP_CMD:
+          if (readTempLM75BD(config.devAddr, &temp) != ERR_CODE_SUCCESS) break; // What should be done if error?
```

Comment:
Log the error if there was an error

Reply:
Also nitpick but I would explicitly use `continue` to make it more clear and also in case something is added after the switch case for some reason in the future (I don't see why this would happen though)

Reply:
I'm not sure I understand, I believe continue doesn't work in a switch case?